.buildkite/commands/lint-localized-strings-format.sh

```diff
@@ -0,0 +1,43 @@
+#!/bin/bash -eu
+
+echo "--- :rubygems: Setting up Gems"
+install_gems
+
+echo "--- :cocoapods: Setting up Pods"
+install_cocoapods
+
+echo "--- :writing_hand: Copy Files"
+mkdir -pv ~/.configure/wordpress-ios/secrets
+cp -v fastlane/env/project.env-example ~/.configure/wordpress-ios/secrets/project.env
+
+echo "--- Lint Localized Strings Format"
+LOGS=logs.txt
+set +e
+set -o pipefail
+bundle exec fastlane generate_strings_file_for_glotpress skip_commit:true | tee $LOGS
+EXIT_CODE=$?
+set -e
+
+echo $EXIT_CODE
+
+if [[ $EXIT_CODE -ne 0 ]]; then
+  # Strings generation finished with errors, extract the errors in an easy-to-find section
+  echo "--- Report genstrings errors"
+  ERRORS=errors.txt
+  echo "Found errors when trying to run \`genstrings\` to generate the \`.strings\` files from \`*LocalizedStrings\` calls:" | tee $ERRORS
+  echo '' >> $ERRORS
+  # Print the errors inline.
+  #
+  # Notice the second `sed` call that removes the ANSI escape sequences that
+  # Fastlane uses to color the output.
+  grep -e "\[.*\].*genstrings:" $LOGS \
+    | sed -e 's/\[.*\].*genstrings: error: /- /' \
+    | sed -e $'s/\x1b\[[0-9;]*m//g' \
```

Comment:
Unfortunately, without that filter, we get a double match for one of the errors.

Take the colored output in the following screenshots:

<img width="1728" alt="image" src="https://user-images.githubusercontent.com/1218433/200803814-0cd58419-7a5d-4608-996f-c02c180cfecf.png">

When run through the code without the ANSI color RegEx, it returns

```
Found errors when trying to run `genstrings` to generate the `.strings` files from `*LocalizedStrings` calls:

- bad entry in file WordPress/Classes/Extensions/Colors and Styles/WPStyleGuide+Search.swift (line = 50): Argument is not a literal string.
- bad entry in file WordPress/Classes/Extensions/Colors and Styles/WPStyleGuide+Search.swift (line = 50): Argument is not a literal string.
- bad entry in file WordPress/Classes/ViewRelated/Aztec/Extensions/Header+WordPress.swift (line = 28): Argument is not a literal string.
```

As far as I can tell, the reason is because of different terminators between the errors printed in line and the ones at the end of the Fastlane execution. With `cat -e`, we can see

```
[05:19:23]: ^[[32m-------------------------------------------------^[[0m$
[05:19:23]: ^[[32m--- Step: ios_generate_strings_file_from_code ---^[[0m$
[05:19:23]: ^[[32m-------------------------------------------------^[[0m$
[05:19:23]: �M-^V� ^[[35mgenstrings: error: bad entry in file WordPress/Classes/Extensions/Colors and Styles/WPStyleGuide+Search.swift (line = 50): Argument is not a literal string.^[[0m$
[05:19:23]: �M-^V� ^[[35mgenstrings: error: bad entry in file WordPress/Classes/ViewRelated/Aztec/Extensions/Header+WordPress.swift (line = 28): Argument is not a literal string.^[[0m$
+------------------+-----------------------------------------+$
|                        ^[[33mLane Context^[[0m                        |$
+------------------+-----------------------------------------+$
| DEFAULT_PLATFORM | ios                                     |$
| PLATFORM_NAME    | ios                                     |$
| LANE_NAME        | ios generate_strings_file_for_glotpress |$
+------------------+-----------------------------------------+$
[05:19:26]: ^[[31mgenstrings: error: bad entry in file WordPress/Classes/Extensions/Colors and Styles/WPStyleGuide+Search.swift (line = 50): Argument is not a literal string.$
genstrings: error: bad entry in file WordPress/Classes/ViewRelated/Aztec/Extensions/Header+WordPress.swift (line = 28): Argument is not a literal string.^[[0m$
[05:19:26]: ^[[31mfastlane finished with errors^[[0m$
```

You can see that the second `bad entry in file WordPress/Classes/Extensions/Colors and Styles/WPStyleGuide+Search.swift (line = 50)...` doesn't have the `^[[0m$` suffix. I'm guessing that's because Fastlane prints all the errors it collected in a single colored expression, but with new lines. The fact that there's only one `[05:19:26]` timestamp annotation for both errors at the end of the logs points to that, too.

I'm sure there's a RegEx we can write to work around all this, but it sounds like we'd be hitting the diminishing return threshold.

Reply:
Ah yeah, I see. Tricky. Nice catch.

Let's keep the filter to remove ANSI code early in the command like you did then.

Besides, even if we managed to keep the colors while still properly removing duplicates, I don't think there's any case where those error lines use multiple styles (e.g. multiple colors / boldness / … within the line to emphasis some words), and instead all the lines seems to all be in a single style each… either all red or all purple. So we wouldn't end up benefit from keeping the coloring much (and even if we wanted those to appear red in the the `stdout` of the dedicated section to make them stand out, we could always re-add the red-color-ANSI-code around it all manually ourselves 🤷 ). So not worth keeping the initial ANSI-codes after all indeed.

Reply:
Tbh, the nicer solution would probably be to have `generate_strings_file_for_glotpress` handle the "write-errors-to-file" capability:
 - Either by providing the right options or indirections from the implementation of the action itself when it calls `genstrings` (something like `genstrings -o Localizable.strings $files 2>genstrings-stderr.log`—or maybe something a bit more convoluted to [use `tee`, but on `stderr` and not `stdout`](https://www.systutorials.com/how-to-make-tee-catch-the-stderr-only-in-linux/), if `genstrings` prints those warnings to `stdout`?—, where the `genstrings-stderr.log` file name would be provided by a `ConfigItem` defaulting to `nil`?), which would probably be the cleanest solution, but would require updating the action itself in release-toolkit to add such support
 - Or by updating the `generate_strings_file_for_glotpress` lane in your `Fastfile` to redirect just the output of the call to `ios_generate_strings_file_from_code` action into a log file (instead of the output of the whole fastlane invocation, which includes that but also all the ceremony with lane switch header, Lane Context table, and summary lines at the end)

And then parse that file to detect and extract the genstrings warnings, without those parsed logs being cluttered by other fastlane logs.

Reply:
@mokagio How do you want to go forward with this PR?
 - Address the above feedback (i.e. move the "write to file redirect" to just the call on `generate_strings_file_for_glotpress` instead of the whole `fastlane` run, or update our action in release toolkit to add some `ConfigItem(key: :errors_log_file)`  to it etc) in this PR?
 - Or merge as is, and address this improvement in a follow-up PR?
 - Or merge as is, and migrate that logic to a dedicated command in `bash-cache` in follow-up PRs, improving it / addressing above feedback in there later too?

Reply:
@AliSoftware thanks for asking.

> Or merge as is, and migrate that logic to a dedicated command in `bash-cache` in follow-up PRs, improving it / addressing above feedback in there later too?

It's, sort of, the best of both worlds. We have something that works and we can ship ASAP, then we can iterate from a centralized place.

I say "sort of", because if we go down the path of updating the toolkit, we either use a default value for the log files or we'd have to update all the clients again.

Still, give this has the possibility of preventing some issue, I'd say let's ship as is ASAP to get more coverage, then improve on the setup.